### PR TITLE
[CWS] retry parent mount

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

* use the proc fallback for parent too
* use container ID cache to retrieve the pid1 which will be use for the fallback
* Do not generate critical resolution error/abnormal path for mount ID that visible only by the kernel (nsfs, etc.)

This PR should reduce the number of abnormal path events.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]
Can you add a bit more description so that we know how to test/QA this ? Otherwise LGTM

**Review by lebauce — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/probe/cgroups_monitor.go [hunk 1]
```diff
@@ -9,58 +9,28 @@
...
 }
 
 // NewCgroupsMonitor returns a new cgroups monitor
-func NewCgroupsMonitor(statsdClient statsd.ClientInterface) *CgroupsMonitor {
+func NewCgroupsMonitor(statsdClient statsd.ClientInterface, cgrouspResolver *resolvers.CgroupsResolver) *CgroupsMonitor {
 	return &CgroupsMonitor{
-		statsdClient: statsdClient,
-		ids:          make(map[string]uint64),
+		statsdClient:    statsdClient,
+		cgrouspResolver: cgrouspResolver,
```
Comment (lebauce) [inline pkg/security/probe/cgroups_monitor.go:34]: nit: cgroupsResolver

File: pkg/security/probe/resolvers/mount_resolver.go [hunk 2]
```diff
@@ -217,41 +237,45 @@ func (mr *MountResolver) Insert(e model.MountEvent) error {
...
 
-	mr.mounts[e.MountID] = &e
+	if mr.minMountID > e.MountID {
+		mr.minMountID = e.MountID
+	}
 }
 
-func (mr *MountResolver) _getParentPath(mountID uint32, cache map[uint32]bool) (string, error) {
+func (mr *MountResolver) _getMountPath(mountID uint32, cache map[uint32]bool) (string, error) {
+	if mountID < mr.minMountID {
```
Comment (lebauce) [inline pkg/security/probe/resolvers/mount_resolver.go:266]: Shouldn't it be mr.IsMountIDValid(mountID) ?

File: pkg/security/probe/resolvers/mount_resolver.go [hunk 3]
```diff
@@ -343,57 +356,117 @@ func (mr *MountResolver) Start(ctx context.Context) {
...
 	defer mr.lock.Unlock()
 
-	mount, err := mr.resolveMount(mountID, pid)
-	if err != nil {
-		return "", "", ErrMountNotFound
+	return mr.resolveMount(mountID, pid, containerID)
+}
+
+func (mr *MountResolver) resolveMount(mountID, pid uint32, containerID string) (*model.MountEvent, error) {
+	if mountID == 0 {
```
Comment (lebauce) [inline pkg/security/probe/resolvers/mount_resolver.go:431]: same as above ?

**Review by lebauce — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

**Pull request merged**